### PR TITLE
OP-1204 Convert to Junit 5

### PR DESCRIPTION
**Pull request by dbmalkovsky**
See OP-1204

### CODE REVIEW

File: src/test/java/org/isf/accounting/Tests.java [hunk 1]
```diff
@@ -469,18 +469,18 @@ public void ioGetDistictsBillItems() throws Exception {
...
 		assertThat(foundBillItem).isNotNull();
 		List<BillItems> billItems = accountingIoOperation.getDistictsBillItems();
 		assertThat(billItems).hasSize(1);
 
 		BillItems billItem = billItems.get(0);
-		assertThat(foundBillItem).isEqualTo(foundBillItem);
 		assertThat(foundBillItem)
-				.isNotEqualTo(TimeTools.getNow())
-				.isEqualTo(billItem);
+			.isEqualTo(foundBillItem)
```
Comment (mwithi) [inline src/test/java/org/isf/accounting/Tests.java:481]: is this equal redundant?

File: src/test/java/org/isf/patient/Tests.java [hunk 2]
```diff
@@ -675,9 +675,9 @@ public void testPatientHashCode() throws Exception {
 	}
 
 	// This test requires access to the opencv java native library
-	@Ignore
+	@Disabled
```
Comment (mwithi) [inline src/test/java/org/isf/patient/Tests.java:678]: Shall we give a try to re-enable all `@Disable` tests?

File: src/test/java/org/isf/therapy/Tests.java [hunk 3]
```diff
@@ -366,15 +366,14 @@ public void testTherapyRowToString() throws Exception {
...
 		TherapyRow foundTherapyRow = therapyIoOperationRepository.findById(id).orElse(null);
-		assertThat(foundTherapyRow).isNotNull();
-
 		assertThat(foundTherapyRow)
-				.isEqualTo(foundTherapyRow)
-				.isNotNull()
-				.isNotEqualTo("someString");
+			.isNotNull()
+			.isEqualTo(foundTherapyRow)
+			.isNotNull()
```
Comment (mwithi) [inline src/test/java/org/isf/therapy/Tests.java:375]: Also here some redundancy

File: src/test/java/org/isf/visits/Tests.java [hunk 4]
```diff
@@ -574,42 +574,42 @@ public void testVisitGetSet() throws Exception {
...
-	public void testVisitEquals() throws Exception {
+	void testVisitEquals() throws Exception {
 		int id = setupTestVisit(false);
 		Visit visit = visitManager.findVisit(id);
 
-		assertThat(visit).isEqualTo(visit);
 		assertThat(visit)
-				.isNotNull()
-				.isNotEqualTo("someString");
+			.isEqualTo(visit)
```
Comment (mwithi) [inline src/test/java/org/isf/visits/Tests.java:610]: also here

File: src/test/java/org/isf/ward/Tests.java [hunk 5]
```diff
@@ -592,20 +592,20 @@ public void testMgrValidationCodeAlreadyExists() throws Exception {
...
 
 	@Test
-	public void testWardEquals() throws Exception {
+	void testWardEquals() throws Exception {
 		Ward ward = testWard.setup(false);
-		assertThat(ward).isEqualTo(ward);
 		assertThat(ward)
-				.isNotNull()
-				.isNotEqualTo("someString");
+			.isEqualTo(ward)
```
Comment (mwithi) [inline src/test/java/org/isf/ward/Tests.java:606]: here too

File: src/test/java/org/isf/dicom/Tests.java [hunk 6]
```diff
@@ -262,25 +262,25 @@ public void testSourceFilesPreloadDicomTest16() throws Exception {
...
 	}
 
-	@Ignore
+	@Disabled
 	// Reason ignored when running CI it generates this error:
 	//    java.awt.HeadlessException:
 	// 	  No X11 DISPLAY variable was set, but this program performed an operation which requires it.
 	@Test
-	public void testSourceFilesConstructorDirectoryNumberOfFiles() throws Exception {
+	void testSourceFilesConstructorDirectoryNumberOfFiles() throws Exception {
```
Comment (mwithi) [inline src/test/java/org/isf/dicom/Tests.java:283]: Shoudl we move this in the GUI project?

File: src/test/java/org/isf/dicom/Tests.java [hunk 7]
```diff
@@ -343,16 +343,16 @@ public void testFileDicomHashCode() throws Exception {
...
-	public void testFileDicomGetDicomType() throws Exception {
+	void testFileDicomGetDicomType() throws Exception {
 		DicomType dicomType = testDicomType.setup(true);
 		FileDicom fileDicom = testFileDicom.setup(dicomType, true);
 		assertThat(fileDicom.getDicomType()).isEqualTo(dicomType);
 	}
 
 	// This test requires access to the opencv native libraries
-	@Ignore
+	@Disabled
```
Comment (mwithi) [inline src/test/java/org/isf/dicom/Tests.java:353]: Should we give a try to all these `@Disabled` tests?

File: src/test/java/org/isf/accounting/Tests.java (hunk 1 above)
Comment (dbmalkovsky) [reply src/test/java/org/isf/accounting/Tests.java:481]: It may look strange but it is there to force executing every statement in the `equals()` method.

Comment (dbmalkovsky) [reply src/test/java/org/isf/accounting/Tests.java:481]: Changed.

File: src/test/java/org/isf/admission/Tests.java [hunk 8]
```diff
@@ -510,50 +539,63 @@ public void testIoGetNextYProg() throws Exception {
...
+	@ParameterizedTest(name = "Test with MATERNITYRESTARTINJUNE={0}")
+	@MethodSource("maternityRestartInJune")
+	void testIoLoadAdmittedPatientNotThere(boolean maternityRestartInJune) throws Exception {
+		GeneralData.MATERNITYRESTARTINJUNE = maternityRestartInJune;
 		assertThat(admissionIoOperation.loadAdmittedPatient(-1)).isNull();
 	}
 
-	@Ignore
-	@Test
+	@Disabled
```
[the remark this answers is not in the document]
Comment (dbmalkovsky) [reply src/test/java/org/isf/admission/Tests.java:583]: Removed `@Disabled` and test runs OK.

File: src/test/java/org/isf/dicom/Tests.java (hunk 6 above)
Comment (dbmalkovsky) [reply src/test/java/org/isf/dicom/Tests.java:283]: I removed the @Disabled and it now runs cleanly.

File: src/test/java/org/isf/dicom/Tests.java (hunk 7 above)
Comment (dbmalkovsky) [reply src/test/java/org/isf/dicom/Tests.java:353]: Seems like it runs cleanly now.

File: src/test/java/org/isf/patient/Tests.java (hunk 2 above)
Comment (dbmalkovsky) [reply src/test/java/org/isf/patient/Tests.java:678]: This needs the native libs which are not in the classpath for unit tests.

File: src/test/java/org/isf/visits/Tests.java (hunk 4 above)
Comment (dbmalkovsky) [reply src/test/java/org/isf/visits/Tests.java:610]: Removed the line as the same test is made a few lines lower in the method.

File: src/test/java/org/isf/therapy/Tests.java (hunk 3 above)
Comment (dbmalkovsky) [reply src/test/java/org/isf/therapy/Tests.java:375]: Removed the duplicate `.isNotNull()`.   The `isEqualTo()` test is done a few lines lower and thus this one is removed.

File: src/test/java/org/isf/ward/Tests.java (hunk 5 above)
Comment (dbmalkovsky) [reply src/test/java/org/isf/ward/Tests.java:606]: Removed the test to itself as it is done later in the method.

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**